storage/upstash/src/index.ts

```diff
@@ -148,102 +148,61 @@ export class UpstashStore extends MastraStorage {
-    if (selectBy?.include?.length) {
-      for (const item of selectBy.include) {
-        messageIds.add(item.id);
-
-        if (item.withPreviousMessages || item.withNextMessages) {
-          // Get the rank of this message in the sorted set
-          const rank = await this.redis.zrank(threadMessagesKey, item.id);
-          if (rank === null) continue;
-
-          // Get previous messages if requested
-          if (item.withPreviousMessages) {
-            const start = Math.max(0, rank - item.withPreviousMessages);
-            const prevIds = await this.redis.zrange(threadMessagesKey, start, rank - 1);
-            prevIds.forEach(id => messageIds.add(id as string));
-          }
+  async getMessages({ threadId }: { threadId: string }): Promise<MessageType[]> {
+    // Use a more specific key pattern that includes the threadId in the correct position
+    const pattern = this.getKey(MastraStorage.TABLE_MESSAGES, { threadId: '*' });
+    const keys = await this.redis.keys(pattern);
+    const messages = await Promise.all(
+      keys.map(async key => {
+        const data = await this.redis.get<MessageType & { _index?: number }>(key);
+        return data;
+      }),
+    );
 
-          // Get next messages if requested
-          if (item.withNextMessages) {
-            const nextIds = await this.redis.zrange(threadMessagesKey, rank + 1, rank + item.withNextMessages);
-            nextIds.forEach(id => messageIds.add(id as string));
+    return (
+      messages
+        .filter(message => message && message.threadId === threadId)
+        .sort((a, b) => {
+          // First try to sort by _index if available
+          if (a!._index !== undefined && b!._index !== undefined) {
+            return a!._index - b!._index;
           }
-        }
-      }
-    }
-
-    // Then get the most recent messages
-    const latestIds = await this.redis.zrange(threadMessagesKey, -limit, -1);
-    latestIds.forEach(id => messageIds.add(id as string));
-
-    // Fetch all needed messages in parallel
-    const messages = (
-      await Promise.all(
-        Array.from(messageIds).map(async id =>
-          this.redis.get<MessageType & { _index?: number }>(this.getMessageKey(threadId, id)),
-        ),
-      )
-    ).filter(msg => msg !== null) as (MessageType & { _index?: number })[];
-
-    // Sort messages by their position in the sorted set
-    const messageOrder = await this.redis.zrange(threadMessagesKey, 0, -1);
-    messages.sort((a, b) => messageOrder.indexOf(a!.id) - messageOrder.indexOf(b!.id));
-
-    // Remove _index before returning
-    return messages.map(({ _index, ...message }) => message as unknown as T);
+          // Fall back to createdAt if _index is not available
+          return new Date(a!.createdAt).getTime() - new Date(b!.createdAt).getTime();
+        })
+        // @ts-ignore
+        .map(({ _index, ...message }) => message as MessageType)
+    ); // Remove _index before returning
```

Comment:
This is weird, not sure how this got in here 🤔all these changes shouldn't be here